### DISCUSSION TITLE
How to check the transport layer settings? [13598]

### DISCUSSION THREAD

**Discussion by carlstand** [eProsima/Fast-DDS#2436] [Q&A]
i'm starting the helloworld example with the following error message:

> Cannot get MAC addresses. Failing back to IP based ID

> [RTPS_TRANSPORT_SHM Error] Failed to create segment f88c521d9daa5d0b: No such process -> Function compute_per_allocation_extra_size
>  [RTPS_MSG_OUT Error] No such process -> Function init

the program still works, but how to make sure i'm really using SHM not UDPv4？is there a way to check the current profiles/configurations fastDDS takes?

i'm running QNX7.1 on a x86_64 arch, thanks!

**Comment by JLBuenoLopez** [comment]
Hi @carlstand,

Sorry for the late reply. There is no easy way to check which of the builtin transports is using your application. You can have more control setting your own transports as explained in our documentation, for instance, [UDP transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport) but you can navigate to the other available transports and look into their corresponding enabling section.

Another way to check the transport layer is to use [Fast DDS Monitor](https://fast-dds-monitor.readthedocs.io/en/latest/) that allows checking the used locators. Also, external tools that capture network traffic can also give you a clue (for example, Wireshark). If you cannot see any user traffic in the wireshark capture but communication is going through, then SHM is probably working. However, seeing the error capture you are providing, I would say that it is more probable that your application is communicating using UDPv4.

I hope this information helps!
